### DISCUSSION TITLE
Control_M not sending line feed

### DISCUSSION THREAD

**Discussion by JoyceBabu** [ghostty-org/ghostty#3510] [Q&A] [answered]
I am using Ghostty on macOS. I always press <kbd>CTRL</kbd>+<kbd>M</kbd> after typing a command instead of pressing the <kbd>return</kbd> key, since it is easier than stretching my little finger.

On Ghosttye <kbd>CTRL</kbd>+<kbd>M</kbd> is clearing the command I have typed.

**Comment by soomtong** [comment]
For me, using `ctrl`+`i` is the same. Of course, the same is true for `ctrl`+`[`. 
I expect this will be fixed soon :)

**Comment by dacheng-gao** [comment]
When I type `CTRL` + `M`, it sends `9;5u`.
When I type `CTRL` + `I`, it sends `5;5u`.
But when I type `CTRL` + `O` (the letter O, not number 0), it sends `Enter`.
`CTRL` + `J` works as `Enter`, too.

But under tmux inside ghostty, `CTRL` + `M` / `O` / `J` works as `Enter`, but `I` not.

I want to know how it works.

**Comment by fitrh** [comment, marked as answer]
Been experiencing this since beta, but I thought it was a shell issue since it only happens inside BusyBox ash, Debian dash, and GNU Bash. fish-shell seems handle it correctly.

As a workaround, I added this to my config

```conf
# HACK: CTRL-M = CR, CTRL-I = Tab, CTRL-[ = Esc in BusyBox ash, Debian dash, and
#       GNU bash. fish-shell seems to handle it correctly, or is it Ghostty
#       handling it incorrectly? Alacritty don't need this hack
keybind = ctrl+m=text:\x0D
keybind = ctrl+i=text:\x09
keybind = ctrl+[=text:\x1B
```

**Reply by mitchellh** [reply]
Details here: https://github.com/ghostty-org/ghostty/discussions/3151#discussioncomment-11678064

**Reply by JoyceBabu** [reply]
@fitrh  Thanks, that worked. Do you know the keybind for Ctrl-d (delete character forward)?

**Reply by fitrh** [reply]
> Do you know the keybind for Ctrl-d (delete character forward)?

@JoyceBabu `7F`, but `CTRL-D` works for me without needing to remap

**Reply by JoyceBabu** [reply]
@fitrh CTRL-D is working without remap in bash, but not in zsh.

Adding `keybind = ctrl+d=text:\x7F` deletes backward like <kbd>bksp</kbd>, not forward like <kbd>del</kbd>.

Where can I find a list of key hash codes?

**Reply by JoyceBabu** [reply]
Backspace should be `\x08`. Is this a mistake?

**Reply by ldemailly** [reply]
it's unfortunate the default isn't to leave Ctrl-M (and ctrl-I and ctrl-[) alone, as it breaks bash etc for no clear benefit, but thanks for the workaround!

**Reply by tk512** [reply]
Funny, this suddenly happened to my terminal (haven't had any issues for months). I googled the symptom and found this discussion that was brought to @mitchellh 's attention.

Any CTRL-C, CTRL-M etc results in 9;5u being repeated.   If you hit Escape it makes a "u".

<img width="1053" height="624" alt="Screenshot 2026-03-25 at 8 33 33 PM" src="https://github.com/user-attachments/assets/d516424a-7c36-4a96-95fc-36a1068e7590" />

If I create a new tab it works again.

I have no idea what I did to trigger this. On version 1.3.1 build 15212 on macOS.
